onnxruntime/core/graph/model.cc

```diff
@@ -319,13 +334,21 @@ static Status LoadModelHelper(const T& file_path, Loader loader) {
       }
     }
   }
-  try {
+
+  auto sub_status = Status::OK();
+  ORT_TRY {
     status = loader(fd);
-  } catch (const std::exception& ex) {
-    GSL_SUPPRESS(es .84)
-    ORT_IGNORE_RETURN_VALUE(Env::Default().FileClose(fd));
-    return Status(ONNXRUNTIME, FAIL, ex.what());
   }
+  ORT_CATCH(const std::exception& ex) {
+    ORT_HANDLE_EXCEPTION([&]() {
+      GSL_SUPPRESS(es .84)
+      ORT_IGNORE_RETURN_VALUE(Env::Default().FileClose(fd));
+      sub_status = Status(ONNXRUNTIME, FAIL, ex.what());
+    });
```

Comment:
Can we just set 'status' instead of having sub_status? We call FileClose below if !status.IsOK so seems like we're just duplicating that here.

Reply:
Same reason as the other place.

---
In reply to: [476241795](https://github.com/microsoft/onnxruntime/pull/4894#discussion_r476241795) [](ancestors = 476241795)